src/arch/cuda/hybridAPI/hapi_impl.cpp

```diff
@@ -143,6 +143,13 @@ void hapiInit(char** argv) {
 #endif
   }
 
+  CmiNodeAllBarrier();
+
+  if (CmiInCommThread()) {
+    // FIXME: Comm. thread sets its device to be the same as worker thread 0
```

Comment:
Is this comment still accurate?  The code below implies otherwise.

Reply:
It is still accurate. If you see the lines 330-332 below, the thread with CmiMyRank 0 saves its device number in GPUManager (which is a CSV variable) which is later used by the comm thread to map to the same device.